pyodide-build/pyodide_build/pywasmcross.py

```diff
@@ -226,14 +226,19 @@ def replay_genargs_handle_dashI(arg: str, target_install_dir: str) -> str | None
         The new argument, or None to delete the argument.
     """
     assert arg.startswith("-I")
-    if (
-        str(Path(arg[2:]).resolve()).startswith(sys.prefix + "/include/python")
-        and "site-packages" not in arg
-    ):
-        return arg.replace("-I" + sys.prefix, "-I" + target_install_dir)
+
     # Don't include any system directories
     if arg[2:].startswith("/usr"):
         return None
+
+    # Replace local Python include paths with the cross compiled ones
+    include_path = str(Path(arg[2:]).resolve())
+    if include_path.startswith(sys.prefix + "/include/python"):
+        return arg.replace("-I" + sys.prefix, "-I" + target_install_dir)
+
+    if include_path.startswith(sys.base_prefix + "/include/python"):
+        return arg.replace("-I" + sys.base_prefix, "-I" + target_install_dir)
```

Comment:
And another question while I am at it. In my case (and it seems in out-of-tree builds in general), `target_install_dir` is `''`, so
`'-I/home/lesteve/micromamba/envs/pyodide-0.24.0a1/include/python3.11'`
gets replaced by
`'-I/include/python3.11'`

`/include/python3.11` does not exist on my machine and in general I would not expect it to exist so I am not too sure about the goal of the replacement besides disabling the include directory ... I guess maybe this replacement is useful for in-tree builds where target_install_dir is not `''`?

Reply:
Oh, indeed we are setting it to an empty string in out-of-tree build.

https://github.com/pyodide/pyodide/blob/e492392f1b7716484a4681b707468fd191818d11/pyodide-build/pyodide_build/out_of_tree/build.py#L27

@hoodmane is it intended? I think it is incorrect.

Reply:
I think I just didn't know what this variable does or how to fill it, and setting it empty didn't break anything so I left it like that. Since something is breaking now it you probably know what should actually go there.

Reply:
Okay, I'll update that part to set the correct target_install_dir. I think it didn't break since we append [cpython include](https://github.com/pyodide/pyodide/blob/e492392f1b7716484a4681b707468fd191818d11/Makefile.envs#L159) all the time.